minimal_scheduling/minimal_sched_fifo_middleware_threads.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Carlos San Vicente
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <pthread.h>
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+/* For this example, we will be configuring the main thread scheduling policy before the node
+ * creation. The thread middleware threads will inherit these settings because they are created
+ * after the scheduling configuration. */
+
+using namespace std::chrono_literals;
+
+class MinimalPublisher : public rclcpp::Node
+{
+public:
+  MinimalPublisher()
+  : Node("minimal_publisher"), count_(0)
+  {
+    publisher_ = this->create_publisher<std_msgs::msg::String>("topic", 10);
+    auto timer_callback =
+      [this]() -> void {
+        auto message = std_msgs::msg::String();
+        message.data = "Hello, world! " + std::to_string(this->count_++);
+        RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
```

Comment:
Is `RCLCPP_INFO` RT safe? My understanding is any sort of logging is not (mutex, syscall, possible dynamic memory allocation). I tend to pass the data (non-string if possible, or maybe just a pointer/reference to a string) from the RT thread to a non RT thread for logging via a ring queue.

Reply:
No, it's not. As a said here https://github.com/ros-realtime/ros2-realtime-examples/pull/2#discussion_r862331816 maybe it's worth clarifying in all the examples that we don't aim to make all the examples real-time compliant.

Reply:
> tend to pass the data (non-string if possible, or maybe just a pointer/reference to a string) from the RT thread to a non RT thread for logging via a ring queue.

Btw, I have a similar example in the [minimal_data_sharing](https://github.com/ros-realtime/ros2-realtime-examples/blob/rolling-experimental/minimal_data_sharing/minimal_lock_free_queue_logging.cpp) package.